packages/app/src/explorer/hooks.ts

```diff
@@ -0,0 +1,49 @@
+import { useKeyboardEvent } from '@react-hookz/web';
+import type { RefObject } from 'react';
+import { useCallback } from 'react';
+
+export function useKeyboardSupport(ref: RefObject<HTMLDivElement>) {
+  const focusNext = useCallback(() => {
+    if (!ref.current) {
+      return;
+    }
+
+    const { activeElement } = document;
+
+    if (
+      !(activeElement instanceof HTMLButtonElement) ||
+      !ref.current.contains(activeElement)
+    ) {
+      return;
+    }
+    const buttonList = [...ref.current.querySelectorAll('button')];
+    const activeIndex = buttonList.indexOf(activeElement);
+
+    if (activeIndex >= 0 && activeIndex < buttonList.length - 1) {
+      buttonList[activeIndex + 1].focus();
+    }
+  }, [ref]);
+
+  const focusPrevious = useCallback(() => {
+    if (!ref.current) {
+      return;
+    }
+
+    const { activeElement } = document;
+    if (
+      !(activeElement instanceof HTMLButtonElement) ||
+      !ref.current.contains(activeElement)
+    ) {
+      return;
+    }
+    const buttonList = [...ref.current.querySelectorAll('button')];
+    const activeIndex = buttonList.indexOf(activeElement);
+
+    if (activeIndex > 0) {
+      buttonList[activeIndex - 1].focus();
+    }
+  }, [ref]);
+
+  useKeyboardEvent('ArrowUp', focusPrevious);
+  useKeyboardEvent('ArrowDown', focusNext);
```

Comment:
Hehe so now, you can actually register the listeners on the explorer with `{ target: ref }`:

```ts
useKeyboardEvent('ArrowUp', focusPrevious, [], { target: ref });
```